src/components/Balance.tsx

```diff
@@ -0,0 +1,63 @@
+import { formatBalance } from "@polkadot/util";
+
+export namespace BalanceStatus {
+  export function idle(): BalanceStatus {
+    return { state: BalanceState.Idle };
+  }
+
+  export function loading(): BalanceStatus {
+    return { state: BalanceState.Loading };
+  }
+
+  export function fetched(value: number): BalanceStatus {
+    return { state: BalanceState.Fetched, value };
+  }
+
+  export function error(message: string): BalanceStatus {
+    return { state: BalanceState.Error, message };
+  }
+}
+
+export enum BalanceState {
+  Idle = "idle",
+  Loading = "loading",
+  Fetched = "fetched",
+  Error = "error",
+}
+
+export type BalanceStatus =
+  | { state: BalanceState.Idle }
+  | { state: BalanceState.Loading }
+  | { state: BalanceState.Fetched; value: number }
+  | { state: BalanceState.Error; message: string };
+
+export function Balance({
+  status,
+  balanceType,
+}: {
+  status: BalanceStatus;
+  balanceType: string;
+}) {
+  switch (status.state) {
+    case "loading":
+      return (
+        <p className="mt-1 text-sm text-gray-400">
```

Comment:
This gray doesn't match the loaded one, I don't think this is necessary in this place